File: Codes in C++/Linked List/Starting Point of a loop.cpp
class Solution {
public:
    ListNode *detectCycle(ListNode *head) {
        ListNode* slow,*fast,*entry;
        slow=head;
        fast=head;
        entry=head;
        while(fast!=NULL&& fast->next!=NULL)
        {
         slow=slow->next;
         fast=fast->next->next;
        if(slow == fast) {
            while(slow != entry) {
                slow = slow->next;
                entry = entry->next;
            }
            return slow;
        }
         }
        return NULL;
    }
};

File: Codes in Python3/Linked List/Starting Point of a loop.py
class Solution:
    def detectCycle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        slow=head
        fast=head
        entry=head
        while fast!=None and fast.next!=None:
            slow=slow.next
            fast=fast.next.next
            if slow==fast:
                while slow!=entry:
                    slow=slow.next
                    entry=entry.next
                if slow==entry:
                    return slow

        return None    